src/fmtargs.h

```diff
@@ -0,0 +1,83 @@
+#define __NARG__(...)  __NARG_I_(__VA_ARGS__,__RSEQ_N())
+#define __NARG_I_(...) __ARG_N(__VA_ARGS__)
+
+/* Define a macro which will call an arbitrary macro appended with a number indicating
+ * the number of arguments it has. */
+#define _VFUNC_(name, n) name##n
+#define _VFUNC(name, n) _VFUNC_(name, n)
+#define VFUNC(func, ...) _VFUNC(func, __NARG__(__VA_ARGS__)) (__VA_ARGS__)
+
+#define COMPACT_FMT(...) VFUNC(COMPACT_FMT_, __VA_ARGS__)
+#define COMPACT_VALUES(...) VFUNC(COMPACT_VALUES_, __VA_ARGS__)
+#define FMT_STRING(initial_string, ...) initial_string COMPACT_FMT(__VA_ARGS__), COMPACT_VALUES(__VA_ARGS__)
```

Comment:
Maybe call it FMTARGS instead of FMT_STRING, to match the header file name? (Or rename the file to `fmt_string.h`).

Do we need `initial_string`? It seems more intuitive without it. For INFO we can just write

```C
info = sdscatprintf(info,
           "# Server\r\n"
           FMT_STRING(
               ...
           ));
```

Reply:
Yeah, naming is a bit crazy. FMT_ARGS makes a bit more sense to me.

Yeah, I think initial_string was a vestigial of the base case needed in another variant (Something like FMT_STRING_1/2/3...). I'll make the updates, and rebase my change ontop of yours.